plugins/outputs/newrelic/newrelic.go

```diff
@@ -40,7 +40,12 @@ func (nr *NewRelic) Connect() error {
 		return fmt.Errorf("InsightKey is a required for newrelic")
 	}
 	var err error
-	nr.harvestor, err = telemetry.NewHarvester(telemetry.ConfigAPIKey(nr.InsightsKey))
```

Comment:
the benefit of using a `ConfigHarvestPeriod` of `0` is that the agent will not do any of its own buffer management of any kind.  This was an issue brought up by @danielnelson in the most recent review.